docs/taskcluster-yml.md

```diff
@@ -0,0 +1,91 @@
+---
+title: .taskcluster.yml
+---
+
+When a push occurs on a monitored repository, the service looks for a
+`.taskcluster.yml` in the root of the repository at the pushed commit. That
+file is read as a YAML document, rendered with
+[JSON-e](https://github.com/taskcluster/json-e), and then used to create the
+resulting tasks.
+
+## File Versions
+
+Every template must contain a top-level property `version: 1`. This is
+consulted before the JSON-e rendering takes place.
+
+There was a version 0 template format, but it was undocumented and was not
+valid YAML.  We shall speak of it no more.
+
+## JSON-e Rendering
+
+The entire YAML file is rendered using
+[JSON-e](https://github.com/taskcluster/json-e). The following context
+variables are provided:
+
+```js
+{
+  // tasksFor (string) - always 'hg-push' for mozilla-taskcluster (this is to
+  // distinguish from other tools that might read the same template, such as
+  // taskcluster-github)
+  tasksFor: 'hg-push',
+
+  push: {
+    // push.owner (string) - the hg user that made the push (not always an email)
+    owner: '..',
+
+    // push.revision (string) - long hg revision hash
+    revision: '..',
+
+    // push.comment (string) - commit comment from the last commit in the push
+    comment: '..'
+
+    // push.pushlog_id (number) - id of this push in the hg pushlog; see
+    // http://mozilla-version-control-tools.readthedocs.io/en/latest/hgmo/pushlog.html#writing-agents-that-consume-pushlog-data
+    pushlog_id: ..,
```

Comment:
If we're going to do underscore_case here and camelCase for `tasksfor` we should probably standardize. I personally prefer the underscores, but either way is probably fine.